### PR TITLE
CB-6862 Zookeeper is not configured with external volume

### PR DESCRIPTION
**Pull request by topolyai5**
Generated template: 
  ```
"services": [
    {
      "refName": "zookeeper",
      "serviceType": "ZOOKEEPER",
      "roleConfigGroups": [
        {
          "refName": "zookeeper-SERVER-BASE",
          "roleType": "SERVER",
          "base": true,
          "configs": [
            {
              "name": "dataDir",
              "value": "/hadoopfs/fs1/zookeeper"
            },
            {
              "name": "dataLogDir",
              "value": "/hadoopfs/fs1/zookeeper"
            }
          ]
        }
      ]
    },..
```
I have tested with bp of https://jira.cloudera.com/browse/CB-5660 and here is the template part:
```
    {
      "refName": "zookeeper",
      "serviceType": "ZOOKEEPER",
      "serviceConfigs": null,
      "roleConfigGroups": [
        {
          "refName": "zookeeper-SERVER-BASE",
          "roleType": "SERVER",
          "base": true,
          "displayName": null,
          "configs": [
            {
              "name": "dataDir",
              "value": "/hadoopfs/fs1/zookeeper",
              "ref": null,
              "variable": null,
              "autoConfig": null
            },
            {
              "name": "dataLogDir",
              "value": "/hadoopfs/fs1/zookeeper",
              "ref": null,
              "variable": null,
              "autoConfig": null
            }
          ]
        }
      ],
      "roles": null,
      "displayName": null,
      "tags": null
    }
```

### CODE REVIEW

File: template-manager-cmtemplate/src/main/java/com/sequenceiq/cloudbreak/cmtemplate/CmHostGroupRoleConfigProviderProcessor.java [hunk 1]
```diff
@@ -124,54 +122,48 @@ private void updateConfigsInTemplate(CmTemplateProcessor templateProcessor, Map<
...
-
-                    // remove original from service
-                    ApiClusterTemplateService service = serviceByRCG.get(configGroupName);
-                    service.getRoleConfigGroups().removeIf(group -> Objects.equals(group.getRefName(), configGroupName));
-
-                    // remove original from instantiator
-                    if (instantiatorRoleConfigGroups != null) {
-                        instantiatorRoleConfigGroups.removeIf(groupInfo -> Objects.equals(groupInfo.getRcgRefName(), configGroupName));
-                    }
+            if (groupCount == 1 || isSharedComponent(templateService.getServiceType(), configGroup.getRoleType())) {
```
Comment (akanto) [inline template-manager-cmtemplate/src/main/java/com/sequenceiq/cloudbreak/cmtemplate/CmHostGroupRoleConfigProviderProcessor.java:128]: This is a complex method, I think we shall increase the coverage

File: template-manager-cmtemplate/src/test/java/com/sequenceiq/cloudbreak/cmtemplate/CmHostGroupRoleConfigProviderProcessorTest.java [hunk 2]
```diff
@@ -226,22 +226,25 @@ public void testGenerateConfigsWhenHdfsJournalNodeNotNull() {
...
 
         Map<String, Map<String, List<ApiClusterTemplateConfig>>> actual = underTest.generateConfigs(templateProcessor, templatePreparator);
-        assertNull(actual.get("hdfs-JOURNALNODE-BASE"));
+        assertNotNull(actual.get("hdfs-JOURNALNODE-BASE"));
     }
 
     @Test
-    public void testGenerateConfigsWhenHdfsJournalNodeIsNotNull() {
-        List<SharedComponent> sharedComponents = List.of();
+    public void testProcessWhenSharedConfigThenAddVolumeConfig() {
```
Comment (akanto) [inline template-manager-cmtemplate/src/test/java/com/sequenceiq/cloudbreak/cmtemplate/CmHostGroupRoleConfigProviderProcessorTest.java:233]: I do not see such test which especially covers the ZK case

File: template-manager-cmtemplate/src/main/java/com/sequenceiq/cloudbreak/cmtemplate/CmHostGroupRoleConfigProviderProcessor.java (hunk 1 above)
Comment (topolyai5) [reply template-manager-cmtemplate/src/main/java/com/sequenceiq/cloudbreak/cmtemplate/CmHostGroupRoleConfigProviderProcessor.java:128]: > This is a complex method, I think we shall increase the test coverage

Test coverage is 91% in this class. I read other tests and the expression is well-tested.

File: template-manager-cmtemplate/src/test/java/com/sequenceiq/cloudbreak/cmtemplate/CmHostGroupRoleConfigProviderProcessorTest.java (hunk 2 above)
Comment (topolyai5) [reply template-manager-cmtemplate/src/test/java/com/sequenceiq/cloudbreak/cmtemplate/CmHostGroupRoleConfigProviderProcessorTest.java:234]: done

**Pull request merged**